apps/extension/src/routes/page/onboarding/set-rpc-endpoint.tsx

```diff
@@ -0,0 +1,94 @@
+import { Card, CardDescription, CardHeader, CardTitle } from '@penumbra-zone/ui/components/ui/card';
+import { FadeTransition } from '@penumbra-zone/ui/components/ui/fade-transition';
+import { RPC_ENDPOINTS } from '../../../shared/rpc-endpoints';
+import { FormEvent, useMemo, useRef, useState } from 'react';
+import { SelectList } from '@penumbra-zone/ui/components/ui/select-list';
+import { Button } from '@penumbra-zone/ui/components/ui/button';
+import { useStore } from '../../../state';
+import { usePageNav } from '../../../utils/navigate';
+import { PagePath } from '../paths';
+import { ServicesMessage } from '@penumbra-zone/types/src/services';
+import { Network } from 'lucide-react';
+
+const randomSort = () => (Math.random() >= 0.5 ? 1 : -1);
+
+export const SetRpcEndpoint = () => {
+  const navigate = usePageNav();
+  const randomlySortedEndpoints = useMemo(() => [...RPC_ENDPOINTS].sort(randomSort), []);
+  const [grpcEndpoint, setGrpcEndpoint] = useState(randomlySortedEndpoints[0]!.url);
+  const setGrpcEndpointInZustand = useStore(state => state.network.setGRPCEndpoint);
+  const customRpcEndpointInput = useRef<HTMLInputElement | null>(null);
```

Comment:
What do you think about moving all this logic into zustand with an RPC slice? My hunch is perhaps we'd want to re-use this within our settings tab in the extension and moving the logic out likely makes this easier.

Reply:
It also reduces the redundancies with `setGrpcEndpointInZustand` & `setGrpcEndpoint`